packages/sitecore-jss/src/graphql-request-client.ts

```diff
@@ -26,14 +27,18 @@ export interface RetryStrategy {
    * @param retries - The number of retries configured.
    * @returns A boolean indicating whether to retry the request.
    */
-  shouldRetry(error: ClientError, attempt: number, retries: number): boolean;
+  shouldRetry(
+    error: ClientError | NodeJS.ErrnoException,
```

Comment:
We are not defining a separate alias name for this just to avoid the breaking change. This way it gives user the flexibility to use `CLientError` type or `NodeJS.ErrnoException` or both as union when they define their `customRetryStrategy`. I tested it and it was working fine.

Reply:
If you try to extend _DefaultStrategy_ or even implement your own that extends _RetryStrategy_
using a changes from _dev_ branch, and after that you apply changes in this PR you might have an error if you will try to access _error.response_ (for example)
```
Property 'response' does not exist on type 'ClientError | ErrnoException'.
  Property 'response' does not exist on type 'ErrnoException'
```
And I think it's expected, since we have a Union of types now instead of a single _ClientError_ type, so it breaks existing implementation